base_layer/wallet/src/output_manager_service/service.rs

```diff
@@ -646,20 +646,31 @@ where
 
         let uo = self.db.fetch_sorted_unspent_outputs().await?;
 
-        match strategy {
-            UTXOSelectionStrategy::Smallest => {
-                for o in uo.iter() {
-                    outputs.push(o.clone());
-                    total += o.value;
-                    // I am assuming that the only output will be the payment output and change if required
-                    fee_without_change = Fee::calculate(fee_per_gram, outputs.len(), 1);
-                    fee_with_change = Fee::calculate(fee_per_gram, outputs.len(), 2);
-
-                    if total == amount + fee_without_change || total >= amount + fee_with_change {
-                        break;
-                    }
-                }
+        let uo = match strategy {
+            UTXOSelectionStrategy::Smallest => uo,
+            // TODO: We should pass in the current height and group
+            // all funds less than the current height as maturity 0
+            UTXOSelectionStrategy::MaturityThenSmallest => {
+                let mut new_uo = uo.clone();
+                new_uo.sort_by(|a, b| match a.features.maturity.cmp(&b.features.maturity) {
```

Comment:
This might be handy here: https://docs.rs/itertools/0.9.0/itertools/trait.Itertools.html#method.group_by

Reply:
Totally, but we didn't seem to have it as a dependency and I was worried to add it